esp-hal-common/src/interrupt/riscv.rs

```diff
@@ -412,9 +414,24 @@ pub unsafe extern "C" fn start_trap_rust_hal(trap_frame: *mut TrapFrame) {
 /// This function is called from an trap handler.
 #[doc(hidden)]
 unsafe fn handle_exception(pc: usize, trap_frame: *mut TrapFrame) {
-    let insn: usize = *(pc as *const _);
+    let insn = if pc % 4 != 0 {
```

Comment:
I'm not sure if you meant to include these changes, but I'm really curious about them: I think I've occasionally seen misaligned access exceptions pointing back here, but IIRC only as a secondary issue (like, when I've smashed the stack or otherwise corrupted the PC). Have I just been lucky, do you think, or is there some other reason the handful of atomic instructions I've got have always fallen on a word boundary?

Also, am I right that this is addressing this part of the TRM:

> The CPU can also access data via the instruction bus, but only in 4-byte aligned manner.

Since `pc` will usually point into one of the ranges only accessible via that bus?

Either way, the changes seem legit to me, I'm just hoping to learn more so I can make a better guess at what problems I've seen that this might help me with.

Reply:
Your're right in that i didn't mean to implement the change here.

I did however run into an issue causing unaligned accesses, since we compile for a target with the C extension, most instructions are 16-bit. This causes some atomic instructions to be 2-byte aligned instead of 4-byte aligned which causes Rust to panic about misaligned access when trying to read the instruction to check whether it is atomic in *some* cases, is my working theory at least.

I ran into this issue while working with a heavily modified version of both the hal and the RT, and i haven't been able to recreate the panic even when inlining assembly specifically, so i *think* this relates to some edge case. If you still have code relying on vanilla crates i'd love to have a look so we can fix this possible bug.

Reply:
It does make sense for this to cause problems in some cases at least, but I've reverted the change, let's save it until there is an example of this causing problems with the vanilla crates. It wasn't relevant to this feature anyway, just a residual from me messing around with the crates in general.